### PR TITLE
MNT: Rebrand SomeViz to Someviz

### LINKED ISSUE #178: Consistent capitalization conventions for specviz/cubeviz/mosviz/jdaviz
We should make the capitalization of the various tools consistent across `jdaviz`.  The following I *think* have been agreed on:

* Specviz/Mosviz/Cubeviz are all capitalized in prose, `Specviz`/`Mosviz`/`Cubeviz` as class names, and `specviz`/`mosviz`/`cubeviz` when used as attributes
* `jdaviz` as a package is lower case.
* In prose, JDAviz vs Jdaviz vs JDAViz still being debated...

We should go through and update any/all doc references for this topic, and also the helper classes will need to be re-named with the capitalization above.

cc @Jenneh @nmearl @dtn5ah @PatrickOgle @rosteen 

[🐱](https://jira.stsci.edu/browse/JDAT-1756)

### LINKED ISSUE #333: Helper classes use old styling
The helpers still use the old styling (SpecViz, CubeViz, MosViz). These should be updated to our new styling (Specviz, Cubeviz, Mosviz)

### LINKED ISSUE #447: What does \f do in Application docstring?
https://github.com/spacetelescope/jdaviz/blob/f79937da9e30c6912f9a2a8de3560fb49fe9c54f/jdaviz/cli.py#L26

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to apply consistent branding across the package. I think this is pretty important to get done early on before we have hundreds notebooks in the wild with old branding.

At the tag-up on 2021-09-10, we decided to not emit deprecation warning yet for SomeViz usage until we can coordinate with JDAT notebooks team (and `spacetelescope/notebooks` too?) to also use the new Someviz classes. See #857 

Fixes #178 and fixes #333

Bonus: Fix #447

Note to self: https://stackoverflow.com/questions/6758963/find-and-replace-with-sed-in-directory-and-sub-directories

TODO:

- [x] Rebrand other viz.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [x] Is a milestone set? Milestone is only currently required for PRs related to Imviz MVP.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/configs/cubeviz/plugins/viewers.py
Comment (pllim) [inline jdaviz/configs/cubeviz/plugins/viewers.py:32]: Actually, I am not sure if this trick would work here given the `@viewer_registry` usage.

If the user does not call this class directly, maybe we can just rename without any backward compatibility?

**Comment by pllim** [comment]
@havok2063 , you should see how much this affects MAST stuff.

**Comment by havok2063** [comment]
@plim oh it does!  this directly impacts MAST.  I would want to try and coordinate when and if this gets merged, if this is included in the tag for the MAST freeze, and if this change is enough to get approval for an update to our OPs environment

**Comment by pllim** [comment]
> this directly impacts MAST

Even with my attempt at backward compatibility?

**Comment by pllim** [comment]
I am hoping this would go in for the upcoming release. Otherwise, renaming will just get more painful if we wait longer.

**Comment by havok2063** [comment]
Keeping the deprecated classes around for a bit definitely helps, and gives us a little wiggle room, regarding deployments.

**Comment by pllim** [comment]
The only thing I cannot really keep is the old _viewer_ classes. I don't think I can register both old and new classes to the same viewer registry. So unless you are calling low level viz stuff directly, I am hoping the impact here is minimal for you. Did you look at the diff?

**Comment by havok2063** [comment]
I had a quick skim through the diff.  We don't use any lower level access at the moment.  Only main class imports, i.e.
```
from jdaviz.app import Application
from jdaviz.core.config import get_configuration
from jdaviz.core.data_formats import get_valid_format
from jdaviz import SpecViz, CubeViz, Specviz2d
```
and accessing the viewer names as written in the `configs` directory to retrieve a viewer's settings.  So as long as the old class names are around for it, I think MAST will continue to work.  But I could a quick test to make sure whenever this is ready to be merged.

**Comment by pllim** [comment]
> But I could a quick test to make sure whenever this is ready to be merged.

I would merge it today if others let me. I think this is important for the upcoming release if we are going to rebrand stuff.

**Comment by havok2063** [comment]
Just tested this.  In order for MAST to continue working in the short term with this PR without changes, we need the deprecated classes available at the top level, i.e. `from jdaviz import SpecViz, CubeViz, Specviz2d`.  Right now they are no longer available.  I think simply adding the old class in each helper init should resolve this.  E.g. in `configs/specviz/__init__.py` modify `from .helper import Specviz` to `from .helper import Specviz, SpecViz`

File: jdaviz/__init__.py
Comment (havok2063) [inline jdaviz/__init__.py:11]:
```suggestion
from jdaviz.configs.specviz import Specviz, SpecViz  # noqa
```

File: jdaviz/__init__.py
Comment (havok2063) [inline jdaviz/__init__.py:13]:
```suggestion
from jdaviz.configs.mosviz import Mosviz, MosViz  # noqa
```

File: jdaviz/__init__.py
Comment (havok2063) [inline jdaviz/__init__.py:14]:
```suggestion
from jdaviz.configs.cubeviz import Cubeviz, CubeViz  # noqa
```

File: jdaviz/configs/cubeviz/__init__.py
Comment (havok2063) [inline jdaviz/configs/cubeviz/__init__.py:2]:
```suggestion
from .helper import Cubeviz, CubeViz  # noqa
```

File: jdaviz/configs/mosviz/__init__.py
Comment (havok2063) [inline jdaviz/configs/mosviz/__init__.py:2]:
```suggestion
from .helper import Mosviz, MosViz  # noqa
```

File: jdaviz/configs/specviz/__init__.py
Comment (havok2063) [inline jdaviz/configs/specviz/__init__.py:2]:
```suggestion
from .helper import Specviz, SpecViz  # noqa
```

**Review by havok2063 — CHANGES_REQUESTED** [review]
See previous comment regarding solution to allow MAST to continue to use this without changes, until we can determine when we can fold these changes into our code.

**Comment by havok2063** [comment]
Tested and works!

**Comment by pllim** [comment]
@havok2063 , please approve if you are happy. Thanks for the review! 😸

**Review by havok2063 — APPROVED** [review]

**Review by rosteen — APPROVED** [review]
Awesome, this is way overdue. Thanks!

**Pull request merged**
